src/LVSS.cpp

```diff
@@ -7,22 +7,99 @@ uint8_t LVSS::getNodeID() {
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Entering idle state");
+    }
+    /* Assigns the VCU signal to the union bit field */
+    this->boardEN.val = VCUBoardSig;
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Battery: %d\r\nHIB: %d\r\nTMS: %d\r\nHUDL: %d\r\nACC: %d\r\nGUB: %d\r\n", boardEN.batt, boardEN.hib, boardEN.tms, boardEN.hudl, boardEN.acc, boardEN.gub);
+
+    /* Turn on boards */
+    powerSwitches[0]->setPowerSwitchStates(boardEN.batt, boardEN.hib);// Turn on Battery and HIB
+    powerSwitches[1]->setPowerSwitchStates(boardEN.tms, boardEN.hudl);// Turn on TMS and HUDL
+    powerSwitches[2]->setPowerSwitchStates(boardEN.acc, boardEN.gub); // Turn on Acc and GUB
+
+    /* Current Checking */
+    PowerSwitchState.battCurrent = powerSwitches[0]->getCurrent(1);
+    PowerSwitchState.tmsCurrent = powerSwitches[1]->getCurrent(1);
+    PowerSwitchState.accCurrent = powerSwitches[2]->getCurrent(1);
+
+    if (PowerSwitchState.battCurrent == -1 || PowerSwitchState.tmsCurrent == -1 || PowerSwitchState.accCurrent == -1) {
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Power Switch Error\r\n");
+        switchFaultstatus = 1;
+    }
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Switch 0 Channel 1: %d\r\nSwitch 1 Channel 1: %d\r\nSwitch 2 Channel 1: %d\r\n", PowerSwitchState.battCurrent, PowerSwitchState.tmsCurrent, PowerSwitchState.accCurrent);
+
+    time::wait(2);// Power switches require the ADC to wait a min of 165 micro seconds before sampling SNS pin again
+
+    PowerSwitchState.hibCurrent = powerSwitches[0]->getCurrent(2);
+    PowerSwitchState.hudlCurrent = powerSwitches[1]->getCurrent(2);
+    PowerSwitchState.gubCurrent = powerSwitches[2]->getCurrent(2);
+
+    if (PowerSwitchState.hibCurrent == -1 || PowerSwitchState.hudlCurrent == -1 || PowerSwitchState.gubCurrent == -1) {
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Power Switch Error\r\n");
+        switchFaultstatus = 1;
+    }
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Switch 0 Channel 2: %d\r\nSwitch 1 Channel 2: %d\r\nSwitch 2 Channel 2: %d\r\n", PowerSwitchState.hibCurrent, PowerSwitchState.hudlCurrent, PowerSwitchState.gubCurrent);
+
+    PowerSwitchState.switch0Temp = powerSwitches[0]->getTempandFault();
+    PowerSwitchState.switch1Temp = powerSwitches[1]->getTempandFault();
+    PowerSwitchState.switch2Temp = powerSwitches[2]->getTempandFault();
+
+    if (PowerSwitchState.switch0Temp == -1 || PowerSwitchState.switch1Temp == -1 || PowerSwitchState.switch2Temp == -1) {
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Power Switch Error\r\n");
+        switchFaultstatus = 1;
+    }
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Switch 0 Temperature: %d\r\nSwitch 1 Temperature: %d\r\nSwitch 2 Temperature: %d\r\n", PowerSwitchState.switch0Temp, PowerSwitchState.switch1Temp, PowerSwitchState.switch2Temp);
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Vicor Current: %d\r\n", acs71240.readCurrent());
+}
```

Comment:
This appears to be missing the logic to shut off a switch when a fault occurs and keep it off until the VCU commands it back on.